client/src/pages/HomePage/HomePage.tsx

```diff
@@ -0,0 +1,124 @@
+import { FormEventHandler, useState } from 'react';
+import { Redirect } from 'react-router-dom';
+import clsx from 'clsx';
+import { Box, Grid, Typography, TextField, Button } from '@material-ui/core';
+import { DatePicker } from '@mui/lab';
+import { ParseableDate } from '@mui/lab/internal/pickers/constants/prop-types';
+import { generateQueryString } from '../../helpers/queryStringHelpers';
+import useStyles from './useStyles';
+
+export default function HomePage(): JSX.Element {
+  const [address, setAddress] = useState('');
+  const [dropInDate, setDropInDate] = useState<ParseableDate<undefined>>(null);
+  const [dropOffDate, setDropOffDate] = useState<ParseableDate<undefined>>(null);
+  const [redirectLink, setRedirectLink] = useState('');
+
+  const classes = useStyles();
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+    const queryString = generateQueryString({ address, dropInDate, dropOffDate });
```

Comment:
Suggestion: It would be nice to have a validation here that if `dropInDate` has been specified then the `dropOffDate` is mandatory :)

Reply:
@rajivtitus I can add that, however the search logic that is later implemented in the integration checks for cases where only one date is specified, I felt that even with incomplete search parameters the user can still. see reasonable search results. [This is the search I'm talking about](https://github.com/hatchways/team-brioche/pull/76#discussion_r721042406). what do you think?

Reply:
@BerryGIT-ME I guess we can have the search with incomplete parameters (The reason I brought up this validation is because I thought of our use case that a user will not indefinitely drop off his dog haha) 😆